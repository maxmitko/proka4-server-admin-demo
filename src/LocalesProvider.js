import { locales } from 'locales'
import React from 'react';

export const LocalesContext = React.createContext();

export default props => <LocalesContext.Provider value={locales} children={props.children} />