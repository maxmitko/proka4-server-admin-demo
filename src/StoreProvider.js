import React from 'react';
import { applyMiddleware, compose, createStore } from 'redux'
import { StoreContext } from 'redux-react-hook';
import createSagaMiddleware from 'redux-saga'
import thunk from 'redux-thunk';
import reducers from 'store/reducers';
import rootSaga from 'store/sagas'

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    reducers,
    composeEnhancers(applyMiddleware(thunk, sagaMiddleware)),
);

sagaMiddleware.run(rootSaga);

const StoreProvider = props => <StoreContext.Provider value={store} children={props.children} />

export default StoreProvider