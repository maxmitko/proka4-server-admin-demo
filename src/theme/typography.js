
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}