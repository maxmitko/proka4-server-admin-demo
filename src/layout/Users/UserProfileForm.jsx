import Button from 'components/Button'
import TextField from 'components/TextField'
import { LocalesContext } from 'LocalesProvider'
import React, { useContext, useState } from 'react'
import styled from 'styled-components';

const UserProfileForm = props => {

    const { formState, initialData } = props;

    const [state, setState] = useState(initialData);
    const locales = useContext(LocalesContext)

    const onChangeField = name => e => {
        setState({
            ...state,
            [name]: e.target.value,
        })
    };

    const handleSubmitForm = e => {
        e.preventDefault()

        const result = { ...state }
        formState.submitForm(result)
    }

    const { username, fullname, email, money, phone, password, repassword } = state

    return (
        <FormStyled onSubmit={handleSubmitForm}>
            <FieldsWrapper>
                <TextField
                    value={username}
                    onChange={onChangeField('username')}
                    label="Логин"
                    labelPosition="left"
                />
                <TextField
                    value={fullname}
                    onChange={onChangeField('fullname')}
                    label="ФИО"
                    labelPosition="left"
                />
                <TextField
                    value={email}
                    onChange={onChangeField('email')}
                    label="Почта"
                    labelPosition="left"
                />
                <TextField
                    value={money}
                    onChange={onChangeField('money')}
                    label="Кошелек (руб)"
                    labelPosition="left"
                />
                <TextField
                    value={phone}
                    onChange={onChangeField('phone')}
                    label="Телефон"
                    labelPosition="left"
                />
                <TextField
                    type="password"
                    value={password}
                    onChange={onChangeField('password')}
                    label="Пароль"
                    labelPosition="left"
                />
                <TextField
                    type="password"
                    value={repassword}
                    onChange={onChangeField('repassword')}
                    label="Повторите пароль"
                    labelPosition="left"
                />
            </FieldsWrapper>
            <ButtonsWrapper>
                <SaveButton type="submit" color="success">{formState.saveButton === locales.create ? locales.create : locales.save}</SaveButton>
            </ButtonsWrapper>
        </FormStyled>
    )
}

export default UserProfileForm

const FormStyled = styled.form`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`

const FieldsWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    padding: 20px 0;
    
    label {
        min-width: 140px;
        align-self: flex-end;
        justify-content: flex-end;
        margin-right: 30px;
        text-align: right;
    }

    > div {
        margin: 15px 0;
    }
`;

const ButtonsWrapper = styled.div`
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;

    > * {
        margin-left: 20px;
    }
`

const SaveButton = styled(Button)`
    svg {
        fill: white;
    }
`