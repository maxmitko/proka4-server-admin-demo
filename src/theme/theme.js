import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme