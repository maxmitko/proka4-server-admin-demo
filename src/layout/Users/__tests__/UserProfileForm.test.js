import LocalesProvider from 'LocalesProvider'
import StoreProvider from 'StoreProvider'
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'

import UserProfileForm from '../UserProfileForm';

const formState = {
    header: "Редактировать",
    saveButton: "Сохранить",
    submitForm: () => { }
}

const initialData = {
    email: "ovosh@mail.ru",
    fullname: "Моисеева Ульяна Дмитриевна",
    id: 5,
    isActive: 0,
    lastAct: "2019-04-03T19:18:33.000Z",
    money: 1200,
    phone: "",
    regDate: "2019-02-18T22:10:14.000Z",
    username: "sweet",
}

const withContext = ui => (
    <StoreProvider>
        <LocalesProvider>
            <ThemeProvider theme={theme}>
                {ui}
            </ThemeProvider >
        </LocalesProvider>
    </StoreProvider>
)

describe('UserProfileForm', () => {
    it('should render correctly', () => {
        const el = withContext(<UserProfileForm formState={formState} initialData={initialData} />)
        const tree = renderer.create(el).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
