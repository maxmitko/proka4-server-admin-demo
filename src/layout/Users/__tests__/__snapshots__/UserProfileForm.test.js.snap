// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileForm should render correctly 1`] = `
<form
  className="sc-iwsKbI fwbdgH"
  onSubmit={[Function]}
>
  <div
    className="sc-gZMcBi jKxSOa"
  >
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Логин"
      onChange={[Function]}
      value="sweet"
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Логин
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value="sweet"
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="ФИО"
      onChange={[Function]}
      value="Моисеева Ульяна Дмитриевна"
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        ФИО
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value="Моисеева Ульяна Дмитриевна"
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Почта"
      onChange={[Function]}
      value="ovosh@mail.ru"
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Почта
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value="ovosh@mail.ru"
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Кошелек (руб)"
      onChange={[Function]}
      value={1200}
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Кошелек (руб)
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value={1200}
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Телефон"
      onChange={[Function]}
      value=""
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Телефон
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Пароль"
      onChange={[Function]}
      type="password"
      value=""
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Пароль
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM eUVPzl"
      color="default"
      label="Повторите пароль"
      onChange={[Function]}
      type="password"
      value=""
    >
      <label
        className="sc-EHOje gRDODJ"
      >
        Повторите пароль
      </label>
      <div
        className="sc-bxivhb bWaDTw"
        color="default"
        onClick={[Function]}
      >
        <input
          className="sc-gzVnrw jBENIC"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <div
          className="sc-ifAKCX bvcOXd"
          color="default"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-gqjmRU fiXlRS"
  >
    <button
      className="sc-VigVT hkvLFm sc-bdVaJa kbnHqY"
      color="success"
      size="medium"
      type="submit"
    >
      <div
        className="sc-htpNat hostrS"
      >
        Сохранить
      </div>
      <div
        className="rtr-root sc-bwzfXH dsGjGK"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        timeout={
          Object {
            "enter": 500,
            "exit": 500,
          }
        }
      >
        <span />
      </div>
    </button>
  </div>
</form>
`;
